stdlib/public/core/StringUTF16View.swift

```diff
@@ -910,17 +910,38 @@ extension String.UTF16View {
     if remaining == 0 { return crumb }
 
     return _guts.withFastUTF8 { utf8 in
-      var readIdx = crumb._encodedOffset
+      var utf16I = 0
+      var guessedIndex = crumb
+#if SWIFT_STDLIB_ENABLE_VECTOR_TYPES
+      if remaining >= 32 {
```

Comment:
Not sure if 32 is the right threshold here but 16 was a hair slower on our benchmarks